views/js/admin/src/components/common/form/PsSelect.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div class="ps-select">
+    <select
+      class="custom-select form-control"
+      :class="{
+        disabled: options.length === 1 || disabled,
+      }"
+      @change="onChange">
+      <option
+        v-for="(item, index) in options"
+        :key="index"
+        :disabled="options.length === 1 || disabled"
+        :selected="mutableValue
+          ? mutableValue === item.value || index === 0
+          : false"
+        :value="item.value"
+        v-text="item.label" />
+    </select>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api';
+import { disabledProps } from '@/composables/props/disabledProps';
+import { useOptionsProps } from '@/composables/props/useOptionsProps';
+import { useSelectModel } from '@/composables/props/model/useSelectModel';
+
+const { model, props, setup } = useSelectModel();
```

Comment:
ik vind `setup` redelijk vage naam eigenlijk. het geeft weinig weg.

Reply:
Het idee was een beetje om dit als een old school mixin te laten werken, maar ik heb nog iets te veel type issues om hem gewoon uit te kunnen pakken in het component zelf